apps/channels/forms.py

```diff
@@ -532,3 +533,92 @@ class CommCareConnectChannelForm(ExtraFormBase):
         help_text="This is the name of the chatbot that will be displayed to users on CommCare Connect",
         max_length=100,
     )
+
+
+class EmbeddedWidgetChannelForm(ExtraFormBase):
+    """Form for creating embedded chat widget channels"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Initialize widget data attributes
+        self.widget_token = None
+        self.embed_code = None
```

Comment:
The way we've handled this before is to have extra fields in the form which are hidden when creating the form and disabled when editing the form: https://github.com/dimagi/open-chat-studio/blob/33f64319ceb4759d63146f490c34276dfbfd01d1/apps/channels/forms.py#L169-L174

That should avoid all the custom code in the views and template.

Another option is to use a custom form template like we do for document sources: https://github.com/dimagi/open-chat-studio/blob/a94707d0c567fa1326d34fbc302dab3105070dcc/templates/documents/document_source_form_dialog.html#L23-L27

In this case I think a combination of both is probably best. You can use a special form field for the widget token and a custom template to allow you to show the embed code.